### DISCUSSION TITLE
Recommendations for tools to schedule PowerShell scripts on multiple Windows servers

### DISCUSSION THREAD

**Discussion by belgium-1z** [PowerShell/PowerShell#24628] [General]
Hi ya'll,

I’ve been using ScriptRunner, PowerShell Universal, Attune, and PDQ Deploy to schedule and manage PowerShell scripts across multiple Windows servers. They’ve worked well for my needs, but I’m interested in exploring other tools that might offer additional features or better fit specific scenarios.

What tools do you use to schedule and run PowerShell scripts on multiple servers? Are there any platforms or methods you’d recommend for centralized management, security, or improved reliability?

Looking forward to your suggestions!

**Comment by kasini3000** [comment]
Powershell devops automation batch framework《kasini3000》

Open source, free, cross-platform powershell Batch, Parallel, automation tools, specially developed for the shortcomings of ANSIBLE

PowerShell is the best engine,  I use it to make the best car.

* [kasini3000](https://gitee.com/chuanjiao10/kasini3000)    win,linux devops automation batch script framework.(It is similar to ansible) ,Welcome to help translate the project to English.

* [kasini3000_agent_linux](https://gitee.com/chuanjiao10/kasini3000_agent_linux) Shell script,one click install powershell on linux,modify sshd_config for PsRemote.

##### English manual:
https://gitee.com/chuanjiao10/kasini3000/blob/master/docs/kasini3000_manual.xlsx

Control machine: Win, Linux
Controlled machine: Win, Linux
Multiple hybrid clouds, multiple Kubernetes clusters, ci/cd,devops.
100% compatible with “controlled machine scripts”： such as VBS, BAT, bash, AWK, Perl
Multi-process batch and Multi-thread batch by “PoshRSJob” module and sqlite-db
You can use PowerShell runspace multit-hreading in Win + PowerShell v5 by “PoshRSJob”

English tech：
https://www.cnblogs.com/piapia/p/16534765.html


##### Difference between kasini3000 and ANSIBLE:
kasini3000 is script, ansible is YAML.
kasini3000 can transfer powershell objects between local and remote machines each other.

transfer powershell objects support ：win<->win,linux<->linux,win<->linux,linux<->win

Note: powershell commands in azure, aws, vmware, win node machines, will generate objects and return to the master machine.
for linux node , you need to construct the object yourself, here is an example:
c:\ProgramData\kasini3000\node_script
Passing objects through SSH-remote or WinRM is the biggest advantage of PowerShell itself.
